src/ngraph/codegen/file_util.cpp

```diff
@@ -0,0 +1,341 @@
+{
+    string rc = s;
+    auto pos = s.find_last_of('/');
+    if (pos != string::npos)
+    {
+        rc = s.substr(pos + 1);
+    }
+    return rc;
+}
+
+std::string codegen::file_util::get_file_ext(const std::string& s)
+{
+    string rc = get_file_name(s);
+    auto pos = rc.find_last_of('.');
+    if (pos != string::npos)
+    {
+        rc = rc.substr(pos);
+    }
+    else
+    {
+        rc = "";
+    }
+    return rc;
+}
+
+string codegen::file_util::path_join(const string& s1, const string& s2)
+{
+    string rc;
+    if (s2.size() > 0)
+    {
+        if (s2[0] == '/')
+        {
+            rc = s2;
+        }
+        else if (s1.size() > 0)
+        {
+            rc = s1;
+            if (rc[rc.size() - 1] != '/')
+            {
+                rc += "/";
+            }
+            rc += s2;
+        }
+        else
+        {
+            rc = s2;
+        }
+    }
+    else
+    {
+        rc = s1;
+    }
+    return rc;
+}
+
+size_t codegen::file_util::get_file_size(const string& filename)
+{
+    // ensure that filename exists and get its size
+
+    struct stat stats;
+    if (stat(filename.c_str(), &stats) == -1)
+    {
+        throw std::runtime_error("Could not find file: \"" + filename + "\"");
+    }
+
+    return stats.st_size;
+}
+
+void codegen::file_util::remove_directory(const string& dir)
+{
+    struct stat status;
+    if (stat(dir.c_str(), &status) != -1)
+    {
+        iterate_files(dir,
+                      [](const string& file, bool is_dir) {
+                          if (is_dir)
+                          {
+                              rmdir(file.c_str());
+                          }
+                          else
+                          {
+                              remove(file.c_str());
+                          }
+                      },
+                      true);
+        rmdir(dir.c_str());
+    }
+}
+
+void codegen::file_util::remove_file(const string& file)
+{
+    remove(file.c_str());
+}
+
+bool codegen::file_util::make_directory(const string& dir)
+{
+    if (mkdir(dir.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH))
+    {
+        if (errno == EEXIST)
+        {
+            // not really an error, the directory already exists
+            return false;
+        }
+        throw std::runtime_error("error making directory " + dir + " " + strerror(errno));
+    }
+    return true;
+}
+
+string codegen::file_util::make_temp_directory(const string& path)
+{
+    string fname = path.empty() ? file_util::get_temp_directory() : path;
+    string tmp_template = file_util::path_join(fname, "ngraph_XXXXXX");
+    char* tmpname = strdup(tmp_template.c_str());
+
+    string rc;
+    if (mkdtemp(tmpname))
+    {
+        rc = tmpname;
+    }
+
+    free(tmpname);
+    return rc;
+}
+
+std::string codegen::file_util::get_temp_directory()
+{
+    const vector<string> potential_tmps = {"NGRAPH_TMP", "TMPDIR", "TMP", "TEMP", "TEMPDIR"};
```

Comment:
This list seems like overkill.

Reply:
This list is to cover multiple operating systems. I would rather have too many than too few